src/contracts/core/DelegationManager.sol

```diff
@@ -298,11 +301,39 @@ contract DelegationManager is
         IERC20[][] calldata tokens,
         bool[] calldata receiveAsTokens
     ) external onlyWhenNotPaused(PAUSED_EXIT_WITHDRAWAL_QUEUE) nonReentrant {
-        for (uint256 i = 0; i < withdrawals.length; ++i) {
+        uint256 n = withdrawals.length;
+        for (uint256 i; i < n; ++i) {
             _completeQueuedWithdrawal(withdrawals[i], tokens[i], receiveAsTokens[i]);
         }
     }
 
+    /// @inheritdoc IDelegationManager
+    function completeQueuedWithdrawal(
+        IERC20[] calldata tokens,
+        bool receiveAsTokens
+    ) external onlyWhenNotPaused(PAUSED_EXIT_WITHDRAWAL_QUEUE) nonReentrant {
+        EnumerableSet.Bytes32Set storage withdrawalRoots = _stakerQueuedWithdrawalRoots[msg.sender];
+        
+        // NOTE: This will revert if the staker does not have at least 1 pending queued withdrawal.
+        _completeQueuedWithdrawal(
+            queuedWithdrawals[withdrawalRoots.at(withdrawalRoots.length() - 1)], 
+            tokens, 
+            receiveAsTokens
+        );
+    }
+
+    /// @inheritdoc IDelegationManager
+    function completeQueuedWithdrawals(
+        IERC20[][] calldata tokens,
+        bool[] calldata receiveAsTokens,
+        uint256 numToComplete
+    ) external onlyWhenNotPaused(PAUSED_EXIT_WITHDRAWAL_QUEUE) nonReentrant {
+        EnumerableSet.Bytes32Set storage withdrawalRoots = _stakerQueuedWithdrawalRoots[msg.sender];
+        for (uint256 i; i < numToComplete; ++i) {
+            _completeQueuedWithdrawal(queuedWithdrawals[withdrawalRoots.at(i)], tokens[i], receiveAsTokens[i]);
```

Comment:
I believe this will also revert calling `withdrawalRoots.at(i)` if numToComplete is greater than the length of enumerableSet. Can we loop for number of iterations = `min(withdrawalRoots.length(), numToComplete)` to avoid this error.